### PR TITLE
Add various sleep functions

### PR DESCRIPTION
**Pull request by danog**

### CODE REVIEW

**Comment by kelunik** [comment]
I don't see any value in usleep. I'd rather just have a single delay function accepting milliseconds.

**Comment by danog** [comment]
Yeah I didn't think it'd be particularly useful, but I'd really like a sleep function that behaves like PHP sleep (accepting seconds, even if in floating point)

**Comment by trowski** [comment]
Since Amp uses milliseconds for timers, I also prefer a single sleep function accepting milliseconds. Adding ` * 1000` isn't a big deal if you want seconds. Let's keep the time currency consistent.

**Comment by danog** [comment]
OK then

File: lib/functions.php [hunk 1]
```diff
@@ -90,6 +90,18 @@ function asyncCall(callable $callback, ...$args)
...
     }
+
+    /**
+     * Sleeps for the specified number of milliseconds.
+     *
+     * @param int $milliseconds
+     *
+     * @return Promise
+     */
+    function sleep(int $milliseconds): Promise
```
Comment (kelunik) [inline lib/functions.php]: Could you please change the name to `delay` to match `Delayed`?

Comment (danog) [reply lib/functions.php]: Done

Comment (kelunik) [reply lib/functions.php]: I meant changing the function name, not the return type.

**Comment by kelunik** [comment]
Thanks!

**Pull request merged**